src/borg/helpers.py

```diff
@@ -1068,7 +1068,7 @@ class Location:
     ssh_re = re.compile(r"""
         (?P<proto>ssh)://                                   # ssh://
         """ + optional_user_re + r"""                       # user@  (optional)
-        (?P<host>[^:/]+)(?::(?P<port>\d+))?                 # host or host:port
+        (?P<host>([^:/]+|\[[^/]+\]))(?::(?P<port>\d+))?     # host or host:port or [ipv6] or [ipv6]:port
```

Comment:
Enough jokes. I suggest we should use [0-9a-fA-F:\.]+ for ipv6 addresses and maybe  [a-zA-Z0-9-_\.]+ for hosts.